### PR TITLE
Use google osdetector plugin instead of internal gradle class.

### PR DESCRIPTION
**Pull request by maxlambrecht**
Replace the internal class OperatingSystem by the osdetector plugin. 
Upgrade gradle wrapper to 6.7.1

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-all.zip
```
Comment (rturner3) [inline gradle/wrapper/gradle-wrapper.properties:6]: Is this Gradle upgrade required to make this work with the `osdetector` plugin? If not, I would probably send it as a separate PR.

Comment (maxlambrecht) [reply gradle/wrapper/gradle-wrapper.properties]: I reverted the Gradle upgrade in this PR, will send a separate PR.

**Review by rturner3 — APPROVED** [review]

**Comment by rturner3** [comment]
Thanks, @maxlambrecht !
